### PR TITLE
fix "Queue a task" / "in parallel" usage

### LINKED ISSUE #113: List of active times is gone
@domenic, exciting to see https://github.com/whatwg/html/commit/19aad5836e2d128c7c87f152e6e3bc6c4ff32d19 updating the timer related prose/algos in HTML. 

Geolocation was depending on the "list of active timers" as part of the [request position](https://w3c.github.io/geolocation-api/#request-position) steps. I'm wondering if you can provide us with some guidance for integrating with the newly exported "run steps after a timeout"?... also, should we use the new timer task source?  

(I'll slap something together... is it ok if I bug you for review?🙏)

### LINKED ISSUE #114: "Queue a task" / "in parallel" usage appears wrong
https://w3c.github.io/geolocation-api/#request-position contains this sentence:

> Queue a global task using the geolocation task source with global to run the following steps in parallel: 

This doesn't really make any sense. Queuing a task is what you do while in parallel, to get back to the main event loop. But you are already on the main event loop. And you can't queue a task to run something in parallel.

https://html.spec.whatwg.org/#event-loop-for-spec-authors may be helpful here. In general it seems like you need an audit of all the steps in the algorithm to ensure that any that involve waiting or talking to external processes (like a location service or permissions checks) happen in parallel, while any that deal with main-thread objects (like callbacks or the current settings object or any data structures like [[watchTasks]]) happen in an appropriately-queued task.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #114
Closes #113

Really sorry folks, this ended needing more changes than I had initially hoped (I found some additional bugs).  

@reillyeon, um, I don't expect you to review all this... I'm wonder if we should just take it, and then do a "full spec review"?

How should we proceed?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/pull/118.html" title="Last updated on Jan 29, 2022, 12:46 AM UTC (e5481f8)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/118/87a31f6...e5481f8.html" title="Last updated on Jan 29, 2022, 12:46 AM UTC (e5481f8)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -488,6 +488,25 @@ <h3>
...
       </section>
+      <section id="check-permission">
+        <h2>
+          Checking permission to use the API
+        </h2>
+        <p>
+          The <cite>Geolocation API</cite> is a [=default powerful feature=].
+        </p>
+        <p>
+          When <dfn data-local-lt="checking permission">check permission</dfn>
```
Comment (reillyeon) [inline index.html:499]:
```suggestion
          When <dfn data-local-lt="checking permission">checking permission</dfn>
```

File: index.html [hunk 2]
```diff
@@ -488,6 +488,25 @@ <h3>
...
+          to use the API, a user agent MAY suggest time-based [=permission=]
+          [=permission/lifetimes=], such as "24 hours", "1 week", or choose to
+          remember the permission [permission/grant=] indefinitely. However, it
```
Comment (reillyeon) [inline index.html:502]:
```suggestion
          remember the permission [=permission/grant=] indefinitely. However, it
```

File: index.html [hunk 3]
```diff
@@ -854,53 +750,159 @@ <h2>
...
-              <li>[=Call back with error=] |errorCallback| and
-              {{GeolocationPositionError/PERMISSION_DENIED}}.
+              <li>Let |permission| be [=get the current permission state=] of
+              <a>"geolocation"</a>.
+              </li>
+              <li>If |permission| is "denied":
+                <ol>
+                  <li>Stop |timeout|.
+                  </li>
+                  <li>Do the <a href="#permission-deined">user or system denied
```
Comment (reillyeon) [inline index.html:797]:
```suggestion
                  <li>Do the <a href="#permission-denied">user or system denied
```

File: index.html [hunk 4]
```diff
@@ -854,53 +750,159 @@ <h2>
...
+                      <li>Stop the |timeout|.
+                      </li>
+                      <li>Go to <a href="#errors-reasons">dealing with
+                      failures</a>.
+                      </li>
+                      <li>Terminate this algorithm.
+                      </li>
+                    </ol>
+                  </li>
+                  <li>If acquiring the position data form the system succeeds:
```
Comment (reillyeon) [inline index.html:840]:
```suggestion
                  <li>If acquiring the position data from the system succeeds:
```

File: index.html [hunk 5]
```diff
@@ -854,53 +750,159 @@ <h2>
...
+                  taking into consideration the value of
+                  |options|.{{PositionOptions/enableHighAccuracy}} during
+                  acquisition.
+                  </li>
+                  <li>If the |timeout| elapses during acquisition, or acquiring
+                  the device's position results in failure:
+                    <ol>
+                      <li>Stop the |timeout|.
+                      </li>
+                      <li>Go to <a href="#errors-reasons">dealing with
```
Comment (reillyeon) [inline index.html:833]: Queue a task to deal with these failures.

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: index.html [hunk 6]
```diff
@@ -854,53 +750,159 @@ <h2>
...
+            <dl class="switch">
+              <dt id="errors-reasons">
+                Dealing with failures:
+              </dt>
+              <dd>
+                <ul>
+                  <li>If acquiring a position fails, do one of the following
+                  based on the condition that matches the failure:
+                    <dl class="switch">
+                      <dt id="permission-deined">
```
Comment (marcoscaceres) [inline index.html:868]:
```suggestion
                      <dt id="permission-denied">
```

File: index.html (hunk 5 above)
Comment (marcoscaceres) [reply index.html:833]: Yep, already done below: "[=Call back with error=]" actually does the task queueing on the "geolocation task source".

**Comment by marcoscaceres** [comment]
Seriously, thanks for going over it @reillyeon! 🙏 Sorry again about all the changes 😅.

But... I.... I THINK WE ARE CR READY!!!! 🎉🎉🎉 🍾🥂

File: index.html (hunk 5 above)
Comment (reillyeon) [reply index.html:843]: Ah, I missed that "call back with error" was a Geolocation API-specific algorithm and not just a generic one to invoke a callback.

File: index.html [hunk 7]
```diff
@@ -638,218 +665,87 @@ <h2>
...
         </h2>
         <p>
-          <dfn>Request position</dfn> by passing a {{PositionCallback}}
-          |successCallback:PositionCallback|, a {{PositionErrorCallback}}`?`
-          |errorCallback:PositionErrorCallback|, {{PositionOptions}}
-          |options:PositionOptions|, a {{boolean}} |repeats:boolean|, and
-          optionally (and only if |repeats| is true) a |previousId:long|.
+          To <dfn>request a position</dfn>, pass a {{PositionCallback}}
+          |successCallback:PositionCallback|, a {{PositionErrorCallback?}}
+          |errorCallback:PositionErrorCallback?|, {{PositionOptions}}
```
Comment (reillyeon) [inline index.html:680]: The type declarations in the method steps for `getCurrentPosition()` and `watchPosition()` should also be updated to use "PermissionErrorCallback?".

**Review by reillyeon — APPROVED** [review]

File: index.html (hunk 7 above)
Comment (marcoscaceres) [reply index.html:680]: Indeed! Fixed.

**Review by reillyeon — APPROVED** [review]

**Pull request merged**